app/components/markdown.tsx

```diff
@@ -99,21 +99,19 @@ export function PreCode(props: { children: any }) {
   );
 }
 
-function escapeDollarNumber(text: string) {
-  let escapedText = "";
-
-  for (let i = 0; i < text.length; i += 1) {
-    let char = text[i];
-    const nextChar = text[i + 1] || " ";
-
-    if (char === "$" && nextChar >= "0" && nextChar <= "9") {
-      char = "\\$";
+function escapeDollarNumber(text: string): string {
+  let isInCodeBlock = false;
+  return text.split('\n').map(line => {
+    if (line.trim() === '```') {
+      isInCodeBlock = !isInCodeBlock;
+      return line;
     }
-
-    escapedText += char;
-  }
-
-  return escapedText;
+    if (!isInCodeBlock) {
+      return line.replace(/(?<!`.*|\\)\$\d+([,.](\d+[,.])?\d+)?(?!.*\$\B)(?!.*`)/g, '\\$&');
```

Comment:
you may need to handle inside single backtick, table as well

Reply:
Thank you for your input. It does handle inside single backticks, unless one used a backtick in one line and another in the next line. Also, it does handle tables as well.

Reply:
How about this?
```
function escapeDollarNumber(text: string): string {
  let isInCode = false;
  return text.split('\n').map(line => {
    if (line.trim() === '```' || line.trim() === '`') {
      isInCode = !isInCode;
      return line;
    }
    if (!isInCode) {
      return line.replace(/(?<!`.*|\\)\$\d+([,.](\d+[,.])?\d+)?(?!.*\$\B)(?!.*`)/g, '\\$&');
    } else {
      return line;
    }
  }).join('\n');
}
```

Reply:
> Thank you for your input. It does handle inside single backticks, unless one used a backtick in one line and another in the next line. Also, it does handle tables as well.

However, it still doesn't handle properly in your [`deployment`](https://4363-vercelpreviewdeployment.pr.nextchat.dev/)

![image](https://github.com/ChatGPTNextWeb/ChatGPT-Next-Web/assets/17626300/496773d7-e006-4ac5-8f1b-e6a0f9a7ec6a)

Reply:
I see. I will try to fix it.
Thank you.

Reply:
Fixed.